.github/workflows/pants.yaml

```diff
@@ -27,13 +27,37 @@ jobs:
     steps:
     - uses: actions/checkout@v2
     - uses: actions/cache@v2
-      id: cache
+      id: cache_pants_setup
       with:
         path: |
           ~/.cache/pants/setup
-          ~/.cache/pants/lmdb_store
+        key: pants-setup-${{ runner.os }}-${{ hashFiles('pants.toml') }}
+        restore-keys: |
+          pants-setup-${{ runner.os }}-${{ hashFiles('pants.toml') }}
+          pants-setup-${{ runner.os }}-
```

Comment:
Restore keys are a double-edged sword. It can speed up CI in the case of a cache miss (`pants.toml` changed) but will also make your cache increase in size over time (as you bump versions, older stale versions of Pants will be still be in the cache).

[There is an upper limit of 10GB](https://github.blog/changelog/2021-11-23-github-actions-cache-size-is-now-increased-to-10gb-per-repository/) at the time of writing, so ever-increasing cache does hurt that,

Reply:
We do document how to prune the cache right below.

Reply:
And this is not really the document for going into that. It's more for giving a general sense of what is cacheable, and how. The problem you mention is a very real but very general one.

Reply:
I don't see us documenting cache pruning. Asleep in not sure that's possible, since the cache is immutable.

Reply:
It's here, in a callout: https://www.pantsbuild.org/docs/using-pants-in-ci

Reply:
Not "pruning" so much as "nuking"